### ISSUE TITLE
Hardcoded FormInput values missing in submission

### ISSUE THREAD

**Issue by crisshaker** [ariakit/ariakit#2451]
### Current behavior

1. FormInput with hardcoded values are not present in submission form values.

https://github.com/ariakit/ariakit/assets/36159205/ebee6dde-be34-4a53-83fd-30d41b96a1cc

### Steps to reproduce the bug

1. Open sandbox: https://codesandbox.io/s/muddy-framework-qcpktf?file=/src/App.tsx
2. Fill the form and submit multiple times
3. See console logs
 

### Expected behavior

1. I expect the set value from `<FormInput name="firstName" value="John" />` to be included in the form data.

**Comment by diegohaz** [comment]
Form currently only supports controlling the values through the store. I'll turn this into a feature request. Could you share more details about your concrete use case and motivation?

**Comment by crisshaker** [comment]
I was making a case at work for how most of Ariakit components have an intuitive and predictable behavior and just work as drop-in replacements for native elements. So it is quite a bummer that this doesn't work. (Side note: I was really looking to replace react-hook-form with Ariakit forms moving forward)

This is a pretty common use case. It is mostly done as hidden inputs:
```tsx
<input name="upcoming_interview_types[]" readOnly hidden value="phone_screen" />.
```

I don't think it is enough to be able to do this only through the store. The same behavior should be possible using just markup:
```tsx
<FormInput name="upcoming_interview_types[]" readOnly hidden value="behavioral_interview" />
````

In short, I expected `<FormInput />` to work just as regular html inputs do at least knowing `<FormInput />` renders an html input under the hood.

I have also attached a screenshot of a real use case in the codebase.

<img width="963" alt="image" src="https://github.com/ariakit/ariakit/assets/36159205/bf945401-998a-406a-bed2-62462ebd740a">

**Comment by diegohaz** [comment]
We should probably support uncontrolled forms too.
